### DISCUSSION TITLE
The byHierarchyMatch failed to intercept a lambda expression

### DISCUSSION THREAD

**Discussion by h-mx** [apache/skywalking#9422] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Java Agent (apache/skywalking-java)

### What happened

Hierarchymatch When I used byHierarchyMatch to intercept a target class via an interface, I found that the implementation class for a lambda expression couldn't be tackled. How do I fix this

### What you expected to happen

This is my implementation. 

  public class HandleInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
  
      private static final String ENHANCE_PARENT_INTERFACE = "com.superxiaobailong.webserver.server.HandlerFunction";
  
      private static final String ENHANCE_METHOD = "handle";
  
      private static final String METHOD_INTERCEPTOR = "org.apache.skywalking.apm.plugin.xbl.webserver.HandleMethodInterceptor";
  
      @Override
      protected ClassMatch enhanceClass() {
          return byHierarchyMatch(ENHANCE_PARENT_INTERFACE);
      }
  
      @Override
      public ConstructorInterceptPoint[] getConstructorsInterceptPoints() {
          return new ConstructorInterceptPoint[0];
      }
  
      @Override
      public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
          return new InstanceMethodsInterceptPoint[]{
                  new InstanceMethodsInterceptPoint() {
                      @Override
                      public ElementMatcher<MethodDescription> getMethodsMatcher() {
                          return named(ENHANCE_METHOD);
                      }
  
                      @Override
                      public String getMethodsInterceptor() {
                          return METHOD_INTERCEPTOR;
                      }
  
                      @Override
                      public boolean isOverrideArgs() {
                          return false;
                      }
                  }
          };
      }
  }

I need to intercept the target class that implements the interface, but when I use a lambda expression, I cannot intercept the implementation class.

    public RouterFunction<ServerResponse> sentMailRouterTwo() {
        return RouterFunctions.route()
                .GET("/sentMail/findById", serverRequest -> sentMailHandler.findById(serverRequest))
                .GET("/sentMail/findListById", new HandlerFunction<ServerResponse>() {
                    @Override
                    public ServerResponse handle(ServerRequest serverRequest) {
                        return sentMailHandler.findListById(serverRequest);
                    }
                })
                .GET("/extension/findByUserIdAndKuaidiCom", extensionHandler::findByUserIdAndKuaidiCom)
                .build();
    }

### How to reproduce

![T}2O{S~ 5QRDD7{(0NQO8H](https://user-images.githubusercontent.com/71714188/182753401-86a8dd1c-ea2b-4096-8153-5c1671792c0d.png)

I want to know how I can solve this problem.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
I don't think there is nothing related to the SkyWalking. 

Agent instrumentation targets compiled classes, not how you write codes. Check the difference between these two inner class files, you will get the idea.
